parquet-column/src/main/java/parquet/vector/VectorizedColumnReaderImpl.java

```diff
@@ -490,7 +489,15 @@ public void readVector(ColumnVector vector) {
       if (!valueRead) {
         binding.readVector(vector);
         valueRead = true;
-        this.pages = null; //processed this batch
+//        this.pages = null; //processed this batch
+        long totalValueCount = 0;
+        for (DataPage page : pages) {
+          totalValueCount += page.getValueCount();
+        }
+        if (vector.size() >= totalValueCount) {
```

Comment:
I think the case is:
vector.size() == totalValueCount
how could it be greater than totalValueCount?